lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -126,9 +113,9 @@ protected function getFilename($id)
     {
         return $this->directory
             . DIRECTORY_SEPARATOR
-            . implode(str_split(hash('sha256', $id), 2), DIRECTORY_SEPARATOR)
+            . substr(hash('sha256', $id), 0, 2)
```

Comment:
> Totally unusable.

Fully agree - we had an old running issue because of that (which can't be fixed in 1.4.x).

This approach puts us at a theoretical limit of 16776960 cache entries (in most cases). Should be enough.